src/core/memory.cpp

```diff
@@ -36,20 +36,15 @@ namespace godot {
 
 void *Memory::alloc_static(size_t p_bytes, bool p_pad_align) {
 #ifdef DEBUG_ENABLED
-	bool prepad = false; // Already pre paded in the engine.
+	bool prepad = true;
```

Comment:
I suppose this means we prepad both if Godot is DEBUG, or if the extension is DEBUG?
I think the reason the engine forces prepad in DEBUG is because it can use the header for debug purposes (although I never checked). Is the same true for GDExtension allocations?

Reply:
> I suppose this means we prepad both if Godot is DEBUG, or if the extension is DEBUG?

Yep!

> I think the reason the engine forces prepad in DEBUG is because it can use the header for debug purposes (although I never checked). Is the same true for GDExtension allocations?

`memnew_arr()` uses the padding to store the array count (regardless of if we're in a debug build or not), which was the source of the memory corruption in my investigation of https://github.com/godotengine/godot-cpp/issues/1820.

I'm not sure what else it's used for, but `memory.hpp` is mostly copied from Godot, so we should be doing all the same things in godot-cpp as are done in Godot

Reply:
> I'm not sure what else it's used for, but memory.hpp is mostly copied from Godot, so we should be doing all the same things in godot-cpp as are done in Godot

I think I see the point from a compat perspective.
But personally, I don't see the appeal of prepad in debug anyway. The prepad information is just the allocation size, which is rarely useful in debugging (and better tracked through professional allocation trackers). And I'm not aware of any godot-cpp machinery that's relying on prepad, other than if `p_prepad=true`, in which case this check is not needed. So I'd rather just pass `p_prepad` through as-is, and leave prepad forcing up to Godot.